src/main/java/com/example/UserModule/repo/UserInsightsRepository.java

```diff
@@ -66,4 +66,11 @@ public interface UserInsightsRepository extends JpaRepository<UserInsights, Inte
   void updateLocationAndDeviceInformation(@Param("id") Integer id, @Param("location") String location,
       @Param("device") String device);
 
+  @Modifying
+  @Transactional
+  @Query("UPDATE UserInsights ui SET ui.firstLoginDate = :firstLoginDate WHERE ui.id = :id")
+  void updateFirstLoginDate(@Param("id") Integer id, @Param("firstLoginDate") Date firstLoginDate);
+
+  @Query("SELECT ui.firstLoginDate FROM UserInsights ui WHERE ui.id = :id")
+  Date getFirstLoginDateUsingId(@Param("id") Integer id);
```

Comment:
**Suggestion:** Consider using Optional<Date> for the return type of getFirstLoginDateUsingId method. This will make it explicit that the method might not always return a value.
```suggestion
  @Query("SELECT ui.firstLoginDate FROM UserInsights ui WHERE ui.id = :id")
  Optional<Date> getFirstLoginDateUsingId(@Param("id") Integer id);
```